### PR TITLE
Add appletBeginBlockingHomeButton() and appletEndBlockingHomeButton()

### PR DESCRIPTION
**Pull request by neonsea**

### CODE REVIEW

**Comment by XorTroll** [comment]
AppletType can also be SystemApplication, I tested it before.

**Comment by neonsea** [comment]
Thanks for the heads up. I was unsure of your indentation style for long lines, so I left it as a half-indent - let me know if that's okay.

File: nx/include/switch/services/applet.h [hunk 1]
```diff
@@ -53,6 +53,10 @@ Result appletCreateManagedDisplayLayer(u64 *out);
 
 Result appletGetDesiredLanguage(u64 *LanguageCode);
 
+Result appletBeginBlockingHomeButton(s64 val);
```
Comment (fincs) [inline nx/include/switch/services/applet.h]: What's this "s64 val"? Is it a bool, is it an enum?

**Comment by XorTroll** [comment]
SWIPC has no documentation about it. but retail apps use 0, and on my tests I used 0 and it worked successfully.

**Comment by neonsea** [comment]
I just tested, and 0 seems to provide the intended behavior, and if this is what official apps use, I think this should be hardcoded to 0. Also added a notice indicating that this only works in regularapps.

**Comment by yellows8** [comment]
The official wrapper funcs calling "BeginBlockingHomeButtonShortAndLongPressed" and "BeginBlockingHomeButton" hard-code the input to 0, however the latter has an additional separate func which passes the input param to the {cmd func}.

For the cmd impl'd here, the input is nanoseconds. Revert the hard-coded input change and update the comment.

**Pull request merged**